wallet/src/de/schildbach/wallet/service/BlockchainServiceImpl.java

```diff
@@ -518,34 +518,36 @@ private void check() {
                         }
 
                         if (!connectTrustedPeerOnly) {
+                            // First use the masternode list that is included
                             try {
-                                peers.addAll(
-                                        Arrays.asList(normalPeerDiscovery.getPeers(services, timeoutValue, timeoutUnit)));
+                                SimplifiedMasternodeList mnlist = org.bitcoinj.core.Context.get().masternodeListManager.getListAtChainTip();
+                                MasternodePeerDiscovery discovery = new MasternodePeerDiscovery(mnlist);
+                                peers.addAll(Arrays.asList(discovery.getPeers(services, timeoutValue, timeoutUnit)));
                             } catch (PeerDiscoveryException x) {
-                                //swallow and continue with another method of connection.
-                                log.info("DNS peer discovery failed: "+ x.getMessage());
-                                if(x.getCause() != null)
-                                    log.info(  "cause:  " + x.getCause().getMessage());
+                                //swallow and continue with another method of connection
+                                log.info("DMN List peer discovery failed: "+ x.getMessage());
                             }
-                            if(peers.size() < 10) {
-                                log.info("DNS peer discovery returned less than 10 nodes.  Adding DMN peers to the list to increase connections");
-                                try {
-                                    SimplifiedMasternodeList mnlist = org.bitcoinj.core.Context.get().masternodeListManager.getListAtChainTip();
-                                    MasternodePeerDiscovery discovery = new MasternodePeerDiscovery(mnlist);
-                                    peers.addAll(Arrays.asList(discovery.getPeers(services, timeoutValue, timeoutUnit)));
-                                } catch (PeerDiscoveryException x) {
-                                    //swallow and continue with another method of connection
-                                    log.info("DMN List peer discovery failed: "+ x.getMessage());
 
+                            if(peers.size() < 16) {
```

Comment:
Since `16` is used twice, it could be stored in a variable to ensure consistent usage through the code.

Reply:
Good Call.

Reply:
Fixed here:  https://github.com/dashevo/dash-wallet/pull/518/commits/12d121db2d55fb6ba109ca5f6b81bf0de8355a0d